nni/runtime/log.py

```diff
@@ -7,14 +7,149 @@
 import sys
 from datetime import datetime
 from io import TextIOBase
-from logging import FileHandler, Formatter, Handler, StreamHandler
+from logging import FileHandler, Formatter, Handler, LogRecord, Logger, StreamHandler
 from pathlib import Path
+import string
 from typing import Optional
 
 import colorama
 
 from .env_vars import dispatcher_env_vars, trial_env_vars
 
+_global_logging_enabled: bool = False
+
+_root_logger: Logger = logging.getLogger()
+logging.getLogger('nni').setLevel(logging.INFO)
+
+_handlers: dict[str, Handler] = {}
+
+colorama.init()
+_colorful: dict[str, str] = colorama.Fore.__dict__
+_colorless: dict[str, str] = {key: '' for key in _colorful.keys()}
+
+def start_stdout_logging() -> None:
+    """
+    Register the stdout handler.
+
+    This function should be invoked on importing nni.
+
+    It is safe to call it multiple times.
+    """
+    if '_stdout_' in _handlers:
+        return
+
+    handler = StreamHandler(sys.stdout)
+    handler.addFilter(_nni_filter)
+    handler.setFormatter(_StdoutFormatter())
+
+    _handlers['_stdout_'] = handler
```

Comment:
This is actually `_nni_stdout_`.

Reply:
`_handlers` is a local variable. So I think no bother.